src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Commitment.cs

```diff
@@ -43,5 +46,96 @@ public Commitment()
         public virtual ICollection<Apprenticeship> Apprenticeship { get; set; }
         public virtual ICollection<Message> Message { get; set; }
         public virtual ICollection<TransferRequest> TransferRequest { get; set; }
+
+        public virtual void AddDraftApprenticeship(DraftApprenticeshipDetails draftApprenticeshipDetails, IUlnValidator ulnValidator)
+        {
+            ValidateDraftApprenticeshipDetails(draftApprenticeshipDetails, ulnValidator);
+            var draftApprenticeship = new DraftApprenticeship(draftApprenticeshipDetails, Originator);
+            Apprenticeship.Add(draftApprenticeship);
+        }
+
+        private void ValidateDraftApprenticeshipDetails(DraftApprenticeshipDetails draftApprenticeshipDetails, IUlnValidator ulnValidator)
+        {
+            var errors = new List<DomainError>();
+            errors.AddRange(BuildNameValidationFailures(draftApprenticeshipDetails));
+            errors.AddRange(BuildEndDateValidationFailures(draftApprenticeshipDetails));
+            errors.AddRange(BuildCostValidationFailures(draftApprenticeshipDetails));
+            errors.AddRange(BuildReferenceValidationFailures(draftApprenticeshipDetails));
+                
+            errors.ThrowIfAny();
```

Comment:
Could we add an extension method to List<DomainError> (or ICollection<..>) to avoid the ToList() in this extension method since it practice it will almost certainly always be a list?

Reply:
Done